biz.aQute.bndlib/src/aQute/bnd/http/HttpClient.java

```diff
@@ -881,4 +882,24 @@ private boolean isUpdateInfo(int code, URLConnection con) {
 		}
 
 	}
+
+	/**
+	 * Validate a URI to see if it is supported by this client
+	 *
+	 * @param u the uri
+	 * @return null if ok, otherwise a reason why it is invalid
+	 */
+	public String validateURI(URI u) {
```

Comment:
Could this be a `static` method? It does not seem to use any instance members.

Reply:
No, that would make it impossible to add schemes via settings if necessary